src/Core/Managed/Shared/Extensibility/MetricManager.cs

```diff
@@ -0,0 +1,285 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility
+{
+    using System;
+    using System.Collections.Concurrent;
+    using System.Collections.Generic;
+    using System.Globalization;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    using Microsoft.ApplicationInsights.DataContracts;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing;
+
+#if CORE_PCL || NET45 || NET46
+    using TaskEx = System.Threading.Tasks.Task;
+#endif
+
+    /// <summary>
+    /// Metric factory and controller.
+    /// </summary>
+    public sealed class MetricManager : IDisposable
+    {
```

Comment:
So I am still reading this, but I really do not like that this guy is disposable.
What are the circumstances where you create more than one MetricManager?
It seems to only be disposable because of the timer. I suggest to look for ways around that. Maybe I can suggest something after I finish reading everything...